src/Controller/Controller.php

```diff
@@ -222,15 +222,39 @@ class Controller implements EventListenerInterface, EventDispatcherInterface
      * Sets a number of properties based on conventions if they are empty. To override the
      * conventions CakePHP uses you can define properties in your class declaration.
      *
-     * @param \Cake\Network\Request|null $request Request object for this controller. Can be null for testing,
+     * The $config array understands the following keys:
+     *
+     * - request: instance of \Cake\Network\Request object for this controller. Can be null for testing,
      *   but expect that features that use the request parameters will not work.
-     * @param \Cake\Network\Response|null $response Response object for this controller.
-     * @param string|null $name Override the name useful in testing when using mocks.
-     * @param \Cake\Event\EventManager|null $eventManager The event manager. Defaults to a new instance.
-     * @param \Cake\Controller\ComponentRegistry|null $components The component registry. Defaults to a new instance.
+     * - response: instance of \Cake\Network\Response object for this controller.
+     * - name: Override the name useful in testing when using mocks.
+     * - eventManager: Instance of \Cake\Event\EventManager. Defaults to a new instance.
+     * - components: Instance of \Cake\Controller\ComponentRegistry. Defaults to a new instance.
+     * @param array $config Options
      */
-    public function __construct(Request $request = null, Response $response = null, $name = null, $eventManager = null, $components = null)
+    public function __construct($options = [])
```

Comment:
I can't exactly where I say this: Future Proof APIs, Options parameters make growing easy. 

:)